src/manager.js

```diff
@@ -70,6 +71,10 @@ class Manager extends EventEmitter {
   async watch (name, options, callback) {
     options.newJobCheckInterval = options.newJobCheckInterval || this.config.newJobCheckInterval
 
+    const teamQueue = options.batchSize ? null : new PQueue({ concurrency: options.teamConcurrency || 1, timeout: 15 * 60 * 1000 })
```

Comment:
I'm not following why we'd use `Promise.race()`?
The timeout setting in p-queue "finishes" the job after that timeout, triggering the `next` event and so replenishing the queue.

Reply:
Sorry, I'm following now. Promise.race() on a per job basis since publish can have different per job timeout

Reply:
I've updated the branch to use Promise.race based on `job.expiresin`

Reply:
After reviewing this a bit more, this will require jumping over 1 last hurdle.  Jobs are retried on explicit failure or expiration from `active` state, but the end state of `expired` is only triggered by an `active` job being marked as such, which means we could most likely get away with kicking the promise from the PQueue in the instance, but not necessarily take action on the promise resolution (resolve or reject, whichever we choose). The maintenance interval also has the ability to be configured, which makes any attempt at "making the decision" probably a bit too high a price to pay in effort and/or length of time.

So, perhaps we just "magic string" a constant and look for that in the failure handler to bypass the explicit failure and let the maintenance handler catch it?  I'm open to other ideas you may have.

Reply:
I've added this in, so to confirm. If a job completes or fails, normal failure or complete handlers are called as expected.
If the timeout triggers, then the job is abandoned and the promise will fall out of the team queue creating more space, but the job will continue running.
There's a bit of a gray zone of what to do with the job if it fails or completes _after_ the timeout.
I've added a `.catch(noop)` so that the job can't raise an unhandled promise rejection if it fails after the timeout.